File: 2020/2.py
passwords = []
with open("2.txt") as file:
    for line in file:
        password = line.split(" ")
        passwords.append(password)

count = 0
for password in passwords:
    letterCount = 0
    for letter in password[3]:
        if letter == password[2]:
            letterCount += 1
    if int(password[0]) <= letterCount <= int(password[1]):
        count += 1
print(count)

count = 0
for password in passwords:
    if password[3][int(password[0])-1] == password[2] or password[3][int(password[1])-1] == password[2]:
        if not (password[3][int(password[0])-1] == password[2] and password[3][int(password[1])-1] == password[2]):
            count += 1
print(count)

File: 2020/Lua/2.lua
local count = 0
local count2 = 0
for line in io.lines("2.txt") do
  local low, up, letter, str = line:match("(%d+)-(%d+) (%l): (%l+)")
  if (str:sub(low, low) == letter and str:sub(up, up) ~= letter) or (str:sub(low, low) ~= letter and str:sub(up, up) == letter)then
    count2 = count2 + 1
  end
  local c = 0
  for char in str:gmatch(".") do
    if char == letter then
      c = c + 1
    end
  end
  if c >= tonumber(low) and c <= tonumber(up) then
    count = count + 1
  end
end
print(count)
print(count2)
